nltepop.cc

```diff
@@ -730,27 +769,86 @@ void set_element_pops_lte(const int nonemptymgi, const int element) {
   gsl_vector_mul(popvec, pop_normfactor_vec);
   // popvec will be used contains the real population densities
 
+  size_t row_ground_state = 0;
   for (size_t row = 0; row < nlte_dimension; row++) {
     double recovered_balance_vector_elem = 0.;
     gsl_vector_const_view row_view = gsl_matrix_const_row(rate_matrix, row);
     gsl_blas_ddot(&row_view.vector, &x, &recovered_balance_vector_elem);
 
-    const auto [ion, level] = get_ion_level_of_nlte_vector_index(row, element);
-
+    const auto [ion, level] = get_ion_level_of_nlte_vector_index(row, element, first_ion_used, nions_used);
+    if (level == 0) {
+      row_ground_state = row;
+    }
     // printout("index %4d (ionstage %d level%4d): residual %+.2e recovered balance: %+.2e normed pop %.2e pop %.2e
     // departure ratio %.4f\n",
     //          row,get_ionstage(element,ion),level, gsl_vector_get(residual_vector,row),
     //          recovered_balance_vector_elem, gsl_vector_get(x,row),
     //          gsl_vector_get(popvec, row),
     //          gsl_vector_get(x, row) / gsl_vector_get(x,get_nlte_vector_index(element,ion,0)));
 
-    if (gsl_vector_get(popvec, row) < 0.0) {
-      printout(
-          "  WARNING: NLTE solver gave negative population to index %zu (Z=%d ionstage %d level %d), pop = %g. "
-          "Replacing with LTE pop of %g\n",
-          row, get_atomicnumber(element), get_ionstage(element, ion), level,
-          gsl_vector_get(&x, row) * gsl_vector_get(pop_normfactor_vec, row), gsl_vector_get(pop_normfactor_vec, row));
-      gsl_vector_set(popvec, row, gsl_vector_get(pop_normfactor_vec, row));
+    if (STRICT_POPULATION_CHECKING) {
```

Comment:
Are you sure that you preserved the old negative population check when STRICT_POPULATION_CHECKING is false?